### PR TITLE
Add stream serialization to ReloadAnalyzersResponse

### LINKED ISSUE #44383: ReloadAnalyzersResponse missing streaming serialization
The new ReloadAnalyzersResponse class, which is returned by the _reload_search_analzyers api, does not implement Writeable or Streamable methods. This means if the response is serialized to another node, the information in the concrete class will be lost (the base class BroadcastResponse would still be deserialized).

Broadcast actions are handled by the coordinating node, and since all nodes can be coordinating, this does not affect the rest layer since the response will be built and immediately serialized to xcontent. However, in 7.x the transport client still exists, where this response would be serialized back to the client, and the reloadDetails would be lost.

### PR DESCRIPTION
**Pull request by cbuescher**
This change adds the missing Streamable support to ReloadAnalyzersResponse that
could be required when using the transport client in 7.x.

Closes #44383

### CODE REVIEW

**Comment by cbuescher** [comment]
@rjernst I'm curious to learn why the missing streamable implementation didn't create errors in tests so far. E.g. in ReloadSynonymAnalyzerIT (which is an ESIntegTestCase), there are checks that test response properties. I was under the impression those integ tests (at least on 7.x) were using the transport client randomly?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/ReloadAnalyzersResponse.java [hunk 1]
```diff
@@ -99,7 +102,36 @@ public static ReloadAnalyzersResponse fromXContent(XContentParser parser) {
         return PARSER.apply(parser, null);
     }
 
-    public static class ReloadDetails {
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        this.reloadDetails.putAll(in.readMap(StreamInput::readString, ReloadDetails::new));
```
Comment (rjernst) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/ReloadAnalyzersResponse.java]: This should be done the ctor that takes StreamInput.  The readFrom should throw UOE.

**Review by rjernst — COMMENTED** [review]
I think this needs to be synced with master, as a refactoring today changes where the deserializing code needs to go.

Regarding the tests, I think that IT likely would be running on one node? I do remember the transport client being randomly used, but I'm not sure why it would not always trip here.

**Comment by cbuescher** [comment]
@rjernst Moved the deserialization code of the respone after merging master and changed ReloadAnalyzerAction to ActionType by following the patter I saw in other BroadcastActions that have been refactored. Let me know if I'm missing something.

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by cbuescher** [comment]
@rjernst anything else to add here?

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by colings86** [comment]
It looks like the backporting is complete here so I have removed the backport pending label while generating 7.4.0 release notes. Please let me know if this is wrong

**Pull request merged**